core/api/views/business_plan_utils.py

```diff
@@ -217,22 +287,37 @@ def parse_bp_file(file, year_start, from_validate=False):
     warnings = []
     for index, row in df.iterrows():
         # parse every row in Excel, get objects by their name
-        agency = agencies.get(strip_str(row["Agency"]))
-        country_name = COUNTRY_NAME_MAPPING.get(row["Country"], row["Country"])
-        country = countries.get(strip_str(country_name))
+        # get error messages
+        agency, country, error_messages = get_error_messages(row, agencies, countries)
+        for error_message in error_messages:
+            if not from_validate:
+                # raise when first error is found and stop parsing entire file
+                raise ValidationError("Data error")
+            errors.append(
+                {
+                    "error_type": "data error",
+                    "row_number": index + 2,
+                    "activity_id": row["Sort Order"],
+                    "error_message": error_message,
+                }
+            )
 
-        # get 'Other' if field is blank or value is not found in db
-        project_type = project_types.get(
-            strip_str(row["Type"]), project_types.get("other")
+        # get 'Other' if value is not found in db, set `None` if field is blank
+        warning_messages = []
+        project_type, warning_messages = get_object(
```

Comment:
In Python, when you pass a mutable object (like a list, dictionary, or a user-defined class instance) to a function, you are passing a reference to that object, not a copy. This means any changes you make to the object inside the function will affect the original object outside the function.

![image](https://github.com/user-attachments/assets/adc8df70-8577-4671-a150-8b240a982e6f)